admin_site/system/views.py

```diff
@@ -651,7 +653,13 @@ def get_context_data(self, **kwargs):
         # Get context from super class
         context = super(ScriptUpdate, self).get_context_data(**kwargs)
         if self.script is not None and self.script.executable_code is not None:
-            context['script_preview'] = self.script.executable_code.read(4096)
```

Comment:
I'm a little nervous about calling `read()` twice, just in case the same file handle is in use behind the scenes. (Also, why _do_ we only read 4096 bytes? It's not like this is a performance-critical corner of the system, and some of our scripts are longer than that...)

Reply:
Some of our scripts have historically been binary files or packed with binary executables. So it's to avoid suddenly reading in several megabytes and trying to display them.

Reply:
Hmm. What about just setting the preview to `<no preview available>` if UTF-8 decoding of the whole file fails, then?

Reply:
Good idea!

Reply:
Resolved in commit 0c5a1da417af2234e8b037565f6f1b8a958b8463.